libparsec/crates/platform_device_loader/src/web/internal.rs

```diff
@@ -0,0 +1,230 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+use std::{path::Path, sync::Arc};
+
+use itertools::Itertools;
+use libparsec_types::{
+    AvailableDevice, DateTime, DeviceAccessStrategy, DeviceFile, DeviceFilePassword, LocalDevice,
+};
+
+use super::error::*;
+
+pub struct Storage {
+    storage: web_sys::Storage,
+}
+
+impl Storage {
+    pub(crate) fn new() -> Result<Self, NewStorageError> {
+        let window = web_sys::window().ok_or(NewStorageError::NoWindow)?;
+        let storage = if cfg!(test) {
+            // Use session storage for tests, So the browser does not try to persist the data.
+            window.session_storage()
+        } else {
+            window.local_storage()
+        }
+        .map_err(NewStorageError::WindowError)
+        .and_then(|v| v.ok_or(NewStorageError::NoLocalStorage))?;
+        Ok(Self { storage })
+    }
+
+    pub(crate) fn list_available_devices(
+        &self,
+        config_dir: &Path,
+    ) -> Result<Vec<AvailableDevice>, ListAvailableDevicesError> {
+        let device_prefix = config_dir
+            .to_str()
+            .expect("The internal path is not a valid UTF-8 string");
```

Comment:
It is not obvious why `config_dir` is guaranteed to be a valid UTF8 path

`config_dir` is a config variable, so it is suggested that it can be set to whatever value by the user the application.

Of course here we are on web so `config_dir` is a fake config value that doesn't get configured (hence why this conversion is safe).
However this feels rather flimsy: if later on we start using `config_dir` for it original purpose, then we would have introduced a bug here that will silently sail to production :/

So given the performance gain is negligible here, I think we'd rather instead use the safe conversion of [to_string_lossy](https://doc.rust-lang.org/std/path/struct.Path.html#method.to_string_lossy)

```suggestion
         let device_prefix = config_dir.to_string_lossy();
```

Reply:
done